File: common/src/main/java/cn/breadnicecat/candycraftce/misc/CGameRules.java
package cn.breadnicecat.candycraftce.misc;

import cn.breadnicecat.candycraftce.utils.CLogUtils;
import net.minecraft.world.level.GameRules;
import org.slf4j.Logger;

/**
 * Created in 2023/12/31 10:48
 * Project: candycraftce
 *
 * @author <a href="https://github.com/Bread-Nicecat">Bread_NiceCat</a>
 * <p>
 */
public class CGameRules {

	private static final Logger LOGGER = CLogUtils.sign();
	public static final GameRules.Key<GameRules.BooleanValue> CARAMEL_PORTAL_WORKS = GameRules.register("doCaramelPortalWorks", GameRules.Category.UPDATES, GameRules.BooleanValue.create(true));

	public static void init() {
	}


}

File: src/main/kotlin/cn/breadnicecat/candycraftce/core/rule/CGameRules.kt
package cn.breadnicecat.candycraftce.core.rule

import cn.breadnicecat.candycraftce.utils.CUtils
import net.fabricmc.fabric.api.gamerule.v1.GameRuleFactory.createBooleanRule
import net.fabricmc.fabric.api.gamerule.v1.GameRuleRegistry
import net.minecraft.world.level.GameRules

/**
 * Created by NiceCat on 2026/2/26.
 * Project: candycraftce
 * @author <a href="https://github.com/BreadNiceCat">Bread_NiceCat</a>
 *
 */
object CGameRules {
    init {
        CUtils.sign()

    }

    val doCaramelPortalWorks: GameRules.Key<GameRules.BooleanValue> = GameRuleRegistry.register(
        "doCaramelPortalWorks",
        GameRules.Category.SPAWNING,
        createBooleanRule(true)
    )
}